source/models/index.py

```diff
@@ -0,0 +1,49 @@
+from abc import ABC, abstractmethod
+import os
+from outputs.exceptions import FileNotFound
+from models.table import Table
+
+class index(ABC):
+    def __init__(self, database_name, table_name):
+        self.indices_path = database_name + "\\" + table_name + "\\indices" 
+        self.database_name = database_name
+        self.table_name = table_name
+
+    def getIndicesNames(self):
+        return os.listdir(self.indices_path)
+
+    def getprimary_keys(self, indexName, index):
+        file_path = self.indices_path + "\\" + indexName + "\\" + index + ".txt"
+        if not os.path.isfile(file_path):
+            raise FileNotFound(message = "this index has not been saved to the database yet")
+        with open(file_path,"r")as primary_keys:
+            primary_keys = primary_keys.read().split(",")
+        return primary_keys
+    
+    def addIndices(self, row_obj):
```

Comment:
add_indices

Reply:
The implementation is too complex here, what I expect the following :- 

```
 list = get_indeces
 list.add(primary_key)
update_indices(index_value)
```

Reply:
this is outdated we solved it in the new version